src/tls/openssl/sni.c

```diff
@@ -0,0 +1,279 @@
+/**
+ * @file openssl/sni.c Server Name Indication API
+ *
+ * Copyright (C) 2022 Commend.com - c.spielberger@commend.com
+ */
+#include <string.h>
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include <openssl/x509.h>
+#include <re_types.h>
+#include <re_fmt.h>
+#include <re_mem.h>
+#include <re_sa.h>
+#include <re_srtp.h>
+#include <re_tls.h>
+#include "tls.h"
+#include "sni.h"
+
+
+#define DEBUG_MODULE "tls"
+#define DEBUG_LEVEL 5
+#include <re_dbg.h>
+
+#include <re_list.h>
+#include <re_hash.h>
+
+
+enum {
+	HANDSH_PROT_SKIP    = 5,
+	CONTENT_TYPE_HANDSH = 22,
+	HANDSH_CLIENT_HELLO = 1,
+	RECORD_LAYER_SKIP   = 4,
+};
+
+struct tls_conn;
+
+static int pl_bin_u8(struct pl *pl, uint8_t *len)
+{
+	if (!pl_isset(pl))
+		return EINVAL;
+
+	*len = (uint8_t) (*pl->p);
+	pl_advance(pl, 1);
+	return 0;
+}
```

Comment:
struct pl is designed for operating on strings.

for this it is probably better to use mbuf